lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -470,6 +528,7 @@ struct VerbatimBuilder {
   Base &base;
   size_t stringBaseSize;
   size_t symbolsBaseSize;
+  SmallString<128> replStr;
```

Comment:
This is rough.  I've had comments going in a few directions, but if this is the way things go, maybe just generate the string on-demand (either the base.string or generate what's needed right there)? Same amount of work for present use case but assuming `getString()` is called once or less than number of times substitutions are added, its' cheaper.

Additionally this keeps VerbatimBuilder small, which helps since that cost is paid for this replStr on every snapshot/copy AFAICT.